packages/frontend/src/components/pagination/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { Dispatch, SetStateAction } from "react";
+import { PaginationBall } from "./pagination-ball";
+import { ReactComponent as ArrowIcon } from "../../assets/arrow.svg";
+import { cva } from "class-variance-authority";
+
+const createArrayBetweenRange = (start: number, end: number) =>
+    Array.from({ length: end - start }, (_, i) => start + i);
+
+const LOT_OF_PAGES = 5;
+
+const arrowsStyles = cva(["cursor-pointer"], {
+    variants: {
+        show: {
+            false: "invisible",
+        },
+    },
+});
+
+export const Pagination = ({
+    totalPages,
+    setCurrentPage,
+    currentPage,
+}: {
+    totalPages: number;
+    currentPage: number;
+    setCurrentPage: Dispatch<SetStateAction<number>>;
+}) => {
+    const isFirstPage = currentPage === 1;
+    const isLastPage = currentPage === totalPages;
+    const hasMoreThanOnePage = totalPages > 1;
+    const hasALotOfPages = totalPages > LOT_OF_PAGES;
+    const isNotFirstPage = currentPage > 1;
+    const isNotLastPage = currentPage < totalPages;
```

Comment:
We can remove `isNotFirstPage` and `isNotFirstPage` here, and use `isFirstPage` and `isLastPage` with a negation `!`, that should work.

I think we can remove `hasALotOfPages` too, and just use the const value when needed.

Also, I'd rename these constants like:

```tsx
    const firstPage = currentPage === 1;
    const lastPage = currentPage === totalPages;
    const moreThanOnePage = totalPages > 1;
```

Reply:
True. Much better. Thanks.

I'll keep the hasALotOfPages variable cause it's simpler than doing the condition and also helps to explain the VISIBLE_PAGES_THRESHOLD variable.

```js
    const hasALotOfPages = totalPages > VISIBLE_PAGES_THRESHOLD;
```

Reply:
fixed: fdd6ea5450d63dad6d1fd228a8acafda08ba2e07